src/App.tsx

```
@@ -1,27 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum SortType {
+  All = 'all',
+  First5 = 'first-five',
+  OnlyRed = 'only-red',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sortType, setSortType] = useState<SortType | null>(null);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleClick = (sort: SortType) => {
+    setSortType(sort);
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  useEffect(() => {
+    if (sortType === SortType.All) {
+      getAll().then(setGoods).catch();
+    }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    if (sortType === SortType.First5) {
+      get5First().then(setGoods).catch();
+    }
+
+    if (sortType === SortType.OnlyRed) {
+      getRedGoods().then(setGoods).catch();
+    }
+  }, [sortType]);
```

This is a good start on error handling, but an empty `.catch()` block isn't sufficient. It prevents the application from crashing but hides the error from both the user and the developer. To properly handle errors as required by checklist item #3, you should at least log the error to the console (e.g., `.catch(console.error)`). A more user-friendly approach would be to set an error state and display a message to the user.